src/Features/Core/Portable/EditAndContinue/Extensions.cs

```diff
@@ -76,9 +77,47 @@ public static bool SupportsEditAndContinue(this Project project)
             => project.Services.GetService<IEditAndContinueAnalyzer>() != null;
 
         // Note: source generated files have relative paths: https://github.com/dotnet/roslyn/issues/51998
-        public static bool SupportsEditAndContinue(this TextDocumentState documentState)
-            => !documentState.Attributes.DesignTimeOnly &&
-               documentState is not DocumentState or DocumentState { SupportsSyntaxTree: true } &&
-               (PathUtilities.IsAbsolute(documentState.FilePath) || documentState is SourceGeneratedDocumentState { FilePath: not null });
+        public static bool SupportsEditAndContinue(this TextDocumentState textDocumentState)
+        {
+            if (textDocumentState.Attributes.DesignTimeOnly)
+            {
+                return false;
+            }
+
+            if (textDocumentState is SourceGeneratedDocumentState { FilePath: not null })
```

Comment:
Strictly speaking, if the document was a source generated document that didn't support syntax trees, this is a change in logic. Not sure how the new "non source outputs" from source generators are going to work, but might be worth adding it for belt-and-braces?

Reply:
Currently all source-gen documents are C#/VB files that support syntax trees:
https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/SourceGeneratedDocumentState.cs,65

We can fix that later if that changes.